Lombiq.Tests.UI/Services/AtataFactory.cs

```diff
@@ -92,10 +89,13 @@ Task<T> FromAsync<T>(Func<BrowserConfiguration, TimeSpan, Task<T>> factory)
             }
             catch (WebDriverException ex)
             {
+                // This warning seems to be a false positive as there is nothing that indicates its validity.
+#pragma warning disable S2589 // Boolean expressions should not be gratuitous
                 if (!ex.Message.ContainsOrdinalIgnoreCase("Cannot start the driver service on") || currentTry >= maxTryCount)
                 {
                     throw;
                 }
+#pragma warning restore S2589 // Boolean expressions should not be gratuitous
 
                 currentTry++;
                 var retryCount = maxTryCount - currentTry;
```

Comment:
Yeah it's false positive, but it can be safely worked around by moving the `currentTry++` before the check. e.g.:
```csharp
                currentTry++;
                var retryCount = maxTryCount - currentTry;

                if (!ex.Message.ContainsOrdinalIgnoreCase("Cannot start the driver service on") || retryCount <= 0)
                {
                    throw;
                }
```

Reply:
In this case shouldn't we adjust the start value of currentTry to 0?

Reply:
If I adjust it to 0 the warning comes back which to me a non-senser :D 
![image](https://github.com/Lombiq/UI-Testing-Toolbox/assets/92299130/adac1c1a-1003-428d-8576-cf841d706be0)

Reply:
If you adjust it to zero then you also have to add a `- 1` to the `retryCount` to retain the same behavior as before.

Reply:
I missed the `retry <=0` change as it was out of the window. rekt.